File: starterpack/blitz2019/java/src/main/java/com/coveo/RandomBot.java
package com.coveo;

import com.coveo.rpc.Action;
import com.coveo.rpc.Agent;
import com.coveo.rpc.Board;
import java.util.Hashtable;
import java.util.List;
import java.util.Random;
import java.util.Vector;

public class RandomBot implements Agent {

  public RandomBot() {
    System.out.println("constructeur");
  }

  @Override
  public Boolean initialize(Hashtable percepts, Vector players, Double time_left) {
    System.out.println("INIT");
    return true;
  }

  @Override
  public Object[] play(Hashtable percepts, int player, int step, Double time_left) {
    System.out.println("PLAY");
    Board board = Board.fromPercepts(percepts);
    List<Action> actions = board.getActions(player);

    Random rand = new Random();
    Action action = actions.get(rand.nextInt(actions.size()));
    return new Object[]{action.actionType, action.coord.i, action.coord.j};
  }
}

File: starterpack/blitz2019/typescript/RandomBot.ts
import {Action} from './Action';
import {Board} from './Board';

export class RandomBot {
    initialize(board: any, players: number[], timeLeft: number | null) {
        console.log('INIT');
    }

    play(percepts: any, player: number, step: number, timeLeft: number | null) {
        console.log('PLAY');

        const board: Board = Board.fromPercepts(percepts);
        const actions: Action[] = board.getActions(player);

        return actions[Math.floor(Math.random() * actions.length)];
    };
}
